### PR TITLE
Finish C# 10 updates

### LINKED ISSUE #27244: Missing entry for C#10
Seems sensible to add an entry for C#10 to this doc so that it can be a simple single reference point
---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 3d252158-0367-77c8-1816-dbc385c5c40b
* Version Independent ID: 575872e5-6273-f8f6-e3e2-2b911cd9d009
* Content: [The history of C# - C# Guide](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-version-history)
* Content Source: [docs/csharp/whats-new/csharp-version-history.md](https://github.com/dotnet/docs/blob/main/docs/csharp/whats-new/csharp-version-history.md)
* Product: **dotnet-csharp**
* Technology: **csharp-whats-new**
* GitHub Login: @erikdietrich
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by BillWagner**
Fixes #27244 

Update the "C# Version history" article.

Also, check keywords pages for any that need to be added for C# 10. In this check, I removed the duplicated table for contextual keywords.

### CODE REVIEW

File: docs/csharp/toc.yml [hunk 1]
```diff
@@ -1052,9 +1052,6 @@ items:
         href: language-reference/keywords/default.md
     - name: Contextual Keywords
       items:
-      - name: Quick reference
-        displayName: contextual keywords
-        href: language-reference/keywords/contextual-keywords.md
```
Comment (adegeo) [inline docs/csharp/toc.yml:1057]: You could, if you wanted, keep this link and point to the index.md#contextual-keywords

**Review by adegeo — APPROVED** [review]

**Review by adegeo — APPROVED** [review]
One minor warning about a bookmark to handle, otherwise GTG

**Comment by BillWagner** [comment]
The bookmark will resolve once #27503 is merged.  That's waiting on a fix in the SDK. (The 6.0.101 build doesn't support the preview generic interfaces.

File: docs/csharp/whats-new/csharp-version-history.md [hunk 2]
```diff
@@ -303,4 +303,36 @@ C# 9 adds many other small features that improve developer productivity, both wr
...
+- [Improvements on lambda expressions](./csharp-10.md#lambda-expression-improvements)
+- [Allow `const` interpolated strings](./csharp-10.md#constant-interpolated-strings)
+- [Record types can seal `ToString()`](./csharp-10.md#record-types-can-seal-tostring)
+- [Improved definite assignment](./csharp-10.md#improved-definite-assignment)
+- [Allow both assignment and declaration in the same deconstruction](./csharp-10.md#assignment-and-declaration-in-same-deconstruction)
+- [Allow `AsyncMethodBuilder` attribute on methods](./csharp-10.md#allow-asyncmethodbuilder-attribute-on-methods)
+- [CallerArgumentExpression attribute](./csharp-10.md#callerargumentexpression-attribute-diagnostics)
+- [Enhanced `#line` pragma](./csharp-10.md#enhanced-line-pragma)
+
+Additional features are available in *preview* mode. You're encouraged to try these features and provide feedback on them. They may change before their final release. In order to use these features, you must [set `<LangVersion>` to `Preview`](../language-reference/compiler-options/language.md#langversion) in your project:
```
Comment (Youssef1313) [inline docs/csharp/whats-new/csharp-version-history.md:325]:
```suggestion
Additional features are available in *preview* mode. You're encouraged to try these features and [provide feedback](https://github.com/dotnet/roslyn/issues/new/choose) on them. They may change before their final release. In order to use these features, you must [set `<LangVersion>` to `Preview`](../language-reference/compiler-options/language.md#langversion) in your project:
```

**Pull request merged**